File: 191-number-of-1-bits/191-number-of-1-bits.py
class Solution(object):
    
    def hammingWeight(self, n):
        """
        :type n: int
        :rtype: int
        """
        binStr = ''
        num = n
        
        while num > 0:
            rem = num % 2
            binStr += str(rem)
            num /= 2
        
        return binStr.count('1')
        

File: 191-number-of-1-bits/191-number-of-1-bits.java
public class Solution {
    // you need to treat n as an unsigned value
    
    public int hammingWeight(int n) {
        String binaryStr = Integer.toBinaryString(n);
        
        int oneCount = 0;
        
        for (int i = 0; i < binaryStr.length(); i++)
            if (binaryStr.charAt(i) == '1')
                oneCount++;
        
        return oneCount;
    }
    
    public int hammingWeight_(int n) {
        int num = n;
        String str = "";
        
        while (num > 0){
            int rem = num % 2;
            str += rem;
            num /= 2;
        }
        
        int oneCount = 0;
        for (int i=0; i<str.length(); i++){
            if (str.charAt(i) == '1')
                oneCount++;
        }
                
        return oneCount;
    }
}